packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -1,111 +1,156 @@
+      ({ index, style }) => {
+        const row = rows[index];
+        prepareRow(row);
+        return (
+          <div {...row.getRowProps({ style })} className={tableStyles.row}>
+            {row.cells.map((cell: Cell, index: number) => (
+              <TableCell
+                key={index}
+                field={data.fields[index]}
+                tableStyles={tableStyles}
+                cell={cell}
+                onCellClick={onCellClick}
+              />
+            ))}
+          </div>
+        );
+      },
+      [prepareRow, rows]
+    );
 
-  const RenderRow = React.useCallback(
-    ({ index, style }) => {
-      const row = rows[index];
-      prepareRow(row);
-      return (
-        <div {...row.getRowProps({ style })} className={tableStyles.row}>
-          {row.cells.map((cell: Cell, index: number) => (
-            <TableCell
-              key={index}
-              field={data.fields[index]}
-              tableStyles={tableStyles}
-              cell={cell}
-              onCellClick={onCellClick}
-            />
-          ))}
-        </div>
-      );
-    },
-    [prepareRow, rows]
-  );
+    return (
+      <div {...getTableProps()} className={tableStyles.table}>
+        <CustomScrollbar>
+          <div style={{ width: `${totalColumnsWidth}px` }}>
+            {!noHeader && (
+              <>
+                {headerGroups.map((headerGroup: HeaderGroup) => {
+                  return (
+                    <div className={tableStyles.thead} {...headerGroup.getHeaderGroupProps()} ref={ref}>
+                      {headerGroup.headers.map((column: Column, index: number) =>
+                        renderHeaderCell(column, tableStyles, data.fields[index], onColumnResize)
+                      )}
+                    </div>
+                  );
+                })}
+              </>
+            )}
+            <FixedSizeList
+              height={height - headerRowMeasurements.height}
+              itemCount={rows.length}
+              itemSize={tableStyles.rowHeight}
+              width={'100%'}
+              style={{ overflow: 'hidden auto' }}
+            >
+              {RenderRow}
+            </FixedSizeList>
+          </div>
+        </CustomScrollbar>
+      </div>
+    );
+  }
+);
 
-  let totalWidth = 0;
+Table.displayName = 'Table';
 
-  for (const headerGroup of headerGroups) {
-    for (const header of headerGroup.headers) {
-      totalWidth += header.width as number;
+function renderHeaderCell(
+  column: any,
+  tableStyles: TableStyles,
+  field?: Field,
+  onColumnResize?: ColumnResizeActionCallback
+) {
+  const headerProps = column.getHeaderProps(column.getSortByToggleProps());
+  if (column.canResize) {
+    headerProps.style.userSelect = column.isResizing ? 'none' : 'auto'; // disables selecting text while resizing
+    column.disableSortBy = column.isResizing; // disables sorting while resizing, unfortunate side effect that the column will not be sortable until re-render
```

Comment:
This hack disables sort during resize, but with the unfortunate side-effect that the column being resized will have sort disabled until table is re-rendered.

Reply:
Fixed by having sort props on a separate div.